manifests/init.pp

```diff
@@ -54,35 +55,41 @@
     }
   }
 
-  if $zebra == true {
-    include quagga::zebra
-  }
+  unless $single_config_file {
```

Comment:
You evaluate the $single_config_file both here and in the classes themselves.  I would recommend evaluating it only here.  Also since puppet 4 evaluates null variables as a true, explicitly state single_config_file == true